src/tabnineChatWidget/workspaceCommands/index.ts

```diff
@@ -0,0 +1,39 @@
+import findSymbolsCommandExecutor from "./commandExecutors/findSymbols";
+
+type WorkspaceCommand = "findSymbols";
+
+export type WorkspaceCommandInstruction = {
+  command: WorkspaceCommand;
+  arg: string;
+};
+
+export type ExecutionResult = {
+  command: WorkspaceCommand;
+  data: string[];
+};
+
+type CommandExecutor = (arg: string) => Promise<string[] | undefined>;
+
+const commandsExecutors: Record<WorkspaceCommand, CommandExecutor> = {
+  findSymbols: findSymbolsCommandExecutor,
+};
+
+export default async function executeWorkspaceCommand(
```

Comment:
maybe you should handle errors here ? you are using Promise.all to map all the command executions.Promise all will fail it has a rejected promise. so either this will aways fulfill the promise without rejecting it or use `allSettled` instead of `all`

Reply:
good point - fixed using allSettled 👍🏻